### ISSUE TITLE
type bool bug

### ISSUE THREAD

**Issue by goldhan** [realm/realm-dart#474]
# 20220420 Flutter Test

Mac mini (M1)

```
flutter doctor                                                                                             
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.10.4, on macOS 12.3.1 21E258 darwin-arm, locale zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.1)
[✓] VS Code (version 1.66.2)
[!] Proxy Configuration
    ! NO_PROXY is not set
[✓] Connected device (3 available)
[!] HTTP Host Availability
    ✗ HTTP host https://cocoapods.org/ is not reachable. Reason: An error occurred while checking the HTTP host: Connection reset by
      peer

! Doctor found issues in 2 categories.
```

[test_demo](https://github.com/FlutterTrip/issues/tree/main/test_flutter)

The same bug as dart

![gif](https://github.com/FlutterTrip/issues/blob/main/test_flutter/8g4wy-w2ocb.gif?raw=true)

---

Hi
ver ^0.2.1+alpha

``` shell
$ dart --version
Dart SDK version: 2.16.2 (stable) (Tue Mar 22 13:15:13 2022 +0100) on "macos_x64"
```

## Step

- `dart pub add realm_dart`
- `dart run realm_dart install`

then run

## bug

![img](https://github.com/FlutterTrip/issues/blob/ea1eca16e2bfc0b98e8852eea5fd7b11d54b94b9/test/1650275208682.jpg?raw=true)

code:

```
realm.write(() {
    /*
      @RealmModel()
class _TestModel {
  @PrimaryKey()
  late int primaryKey = 0;
  bool test1 = true;
  bool test2 = false;
}
    */
    realm.add(TestModel()); // default model
    realm.add(TestModel(primaryKey: 1));
    realm.add(TestModel(test1: false, test2: true, primaryKey: 2));
    realm.add(TestModel(test1: true, test2: false, primaryKey: 3));
  });
```
type  `bool`

添加进去的数据，和最终取出来的相反
The added data is the opposite of what is finally taken out

> code: test1 = false, realm file test1 = true 

[test_demo](https://github.com/FlutterTrip/issues/tree/main/test)

**Comment by nielsenko** [comment]
Hi @goldhan 

Thrilled to see you taking Realm for Dart for a spin! 

I cannot reproduce the issue you see.

I have opened a PR against you repo (https://github.com/FlutterTrip/issues/pull/1), where I fix `cleanRealm` and add some asserts to illustrate that the persisted state is sane.

Could I ask you to run again on your end, and forward any assert failures you may see?

Br

**Comment by goldhan** [comment]
> 

run ... all done!
but ... in Realm Studio 
![image](https://user-images.githubusercontent.com/20206273/164024449-82e60eb1-4b1b-43b7-a8ce-eae4e9b6470c.png)

**Comment by goldhan** [comment]
> 
Hi @nirinchev 
this file   
default.realm
![image](https://user-images.githubusercontent.com/20206273/164166307-fb324ec9-b941-4102-93a2-9c01a2b24193.png)

**Comment by nielsenko** [comment]
Hi @goldhan 

I can reproduce what you see. I have tested the realm with the .NET SDK as well, which agrees with Realm Studio, so this is a bug in the Dart SDK.

I believe we have an issue regarding mapping of property names to meta-data keys. This would explain why we don't see the issue inside the app, as the mapping is consistent between reads and writes.

This will be fixed before the next release, but I'm not sure exactly when that will land.

Br Kasper

**Comment by goldhan** [comment]
> 



> Hi @goldhan
> 
> I can reproduce what you see. I have tested the realm with the .NET SDK as well, which agrees with Realm Studio, so this is a bug in the Dart SDK.
> 
> I believe we have an issue regarding mapping of property names to meta-data keys. This would explain why we don't see the issue inside the app, as the mapping is consistent between reads and writes.
> 
> This will be fixed before the next release, but I'm not sure exactly when that will land.
> 
> Br Kasper

Thanks @nielsenko  
I'm looking forward to it.
Realm performed great in my project testing. 😃

**Comment by nielsenko** [comment]
@goldhan The fix is released with 0.3.0+beta.
